android/src/main/java/com/it_nomads/fluttersecurestorage/FlutterSecureStoragePlugin.java

```diff
@@ -59,25 +65,14 @@ public void initInstance(BinaryMessenger messenger, Context context) {
       }
     }
 
-    /**
-     * This must be run in a separate Thread from an async method to avoid hanging UI thread on
-     * live devices in release mode.
-     * The most convenient place for that appears to be onMethodCall().
-     */
     private void ensureInitStorageCipher() {
-        // Check to avoid unnecessary entry into the synchronized block.
         if (storageCipher == null) {
-            synchronized (this) {
```

Comment:
Why do you remove this sync point ?

Reply:
Because there's only one thread from which the method is ever called. No need to synchronize. That's the whole point of this change.